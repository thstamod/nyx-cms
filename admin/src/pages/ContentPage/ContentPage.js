import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import styled from 'styled-components';
import { Col, Row } from 'react-bootstrap';
import Sidebar from '../../components/Navigation/sidebar/Sidebar';
import GET_DOCUMENT_TYPES from './query';

const FullRow = styled(Row)`
  height: 100%;
  margin-right: 0;
  margin-left: 0;
`;
const FullCol = styled(Col)`
  height: 100%;
  padding-left: 0;
  padding-right: 0;
`;

const ContentPage = () => {
  const { loading, data, error } = useQuery(GET_DOCUMENT_TYPES, {
    errorPolicy: 'all',
  });
  const handleError = (err) =>
    err && (
      <div>
        Bad:
        {err.graphQLErrors.map(({ message }, i) => (
          <span key={i}>{message}</span>
        ))}
      </div>
    );
  if (loading) {
    return <div>Loading...</div>;
  }
  if (data) {
    console.log(data);
  }

  const showSidebar = (d) => <Sidebar data={d} />;
  return (
    <FullRow>
      <FullCol xs={3} md={2}>
        {showSidebar(data)}
      </FullCol>
      <FullCol style={{ overflow: 'auto' }} xs={9} md={10}>
        <h1>Content page</h1>

        {handleError(error)}
      </FullCol>
    </FullRow>
  );
};

export default ContentPage;
