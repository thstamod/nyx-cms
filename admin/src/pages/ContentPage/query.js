import gql from 'graphql-tag';

export default gql`
  query dt {
    documentTypes: getDocunemtTypes {
      _id
      name
      descendants {
        documentType {
          name
          _id
        }
      }
    }
  }
`;
