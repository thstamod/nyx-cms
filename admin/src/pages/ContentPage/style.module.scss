@import '../../scss/base.scss';

.mainContent {
  flex-grow: 1;
}

.text {
  // compose:
  color:red

}

