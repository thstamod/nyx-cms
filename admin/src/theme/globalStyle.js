import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  html,body {
    width:100%;
  height: 100vh;
  display: flex;
  font-family: Karla, serif;
  }
#root {
  width:100%;
  height:100%;
}

  h1,h2,h3,h4,h5 {
    font-family: Rubik, serif;
  }
`;

export default GlobalStyle;
