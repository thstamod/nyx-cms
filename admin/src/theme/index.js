const theme = {
  colors: {
    powderWhite: '#F2F3F4',
    foggyOrange: '#D6804D',
    lightPurple: '#B886A5',
    darkOrange: '#AB5C49',
    darkGrey: '#262728',
  },
  fonts: ['Karla,sans-serif', 'Rubik,sans-serif'],
  fontSizes: {
    small: '1em',
    medium: '2em',
    large: '3em',
  },
};

export default theme;
