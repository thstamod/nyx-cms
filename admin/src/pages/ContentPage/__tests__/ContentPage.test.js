import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, waitFor } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { MockedProvider } from '@apollo/react-testing';
import theme from '../../../theme';
import ContentPage from '../ContentPage';
import GET_DOCUMENT_TYPES from '../../../graphql/getDocumentTypesQuery';
import data from '../../../../__mocks__/_pages/_contentPage';

afterEach(cleanup);

const mocks = [
  {
    request: {
      query: GET_DOCUMENT_TYPES,
    },
    result: {
      data,
    },
  },
];

test('Content Page renders', () => {
  const { asFragment } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider theme={theme}>
        <ContentPage />
      </ThemeProvider>
    </MockedProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});

it('should render loading state initially', () => {
  const { getByText } = render(
    <MockedProvider mocks={[]} addTypename={false}>
      <ThemeProvider theme={theme}>
        <ContentPage />
      </ThemeProvider>
    </MockedProvider>
  );

  const loading = getByText(/Loading.../i);
  expect(loading).toHaveTextContent('Loading...');
});

test('Content Page has header and sidebar', async () => {
  const { getByTestId } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider theme={theme}>
        <ContentPage />
      </ThemeProvider>
    </MockedProvider>
  );
  await waitFor(() => expect(getByTestId('sidebar')).toBeInTheDocument());
});
