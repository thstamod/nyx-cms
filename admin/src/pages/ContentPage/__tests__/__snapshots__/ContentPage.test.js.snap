// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Page renders 1`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
